helm/nginx-ingress-controller-app/templates/controller-deployment.yaml

```diff
@@ -47,6 +47,30 @@ spec:
       serviceAccountName: {{ .Values.controller.name }}
       priorityClassName: system-cluster-critical
       terminationGracePeriodSeconds: {{ .Values.controller.terminationGracePeriodSeconds }}
+      initContainers:
+      # Overall performance improvements
+      # See https://github.com/kubernetes/ingress-nginx/issues/1939
+      - name: {{ .Values.initContainer.name }}
+        command:
+        - sh
+        - -c
+        - |
+          set -o errexit
+          set -o xtrace
+          until sudo sysctl -w net.core.somaxconn=32768
+          do
+            sleep 60s
```

Comment:
Why is this a loop? Taking potentially several minutes to start the pod doesn't feel ideal.

Reply:
that is a good point - maybe sleeping for 5 seconds would make more sense. the idea behind the loop is to not allow the initcontainer to fail silently - with this approach it's pretty obvious the issue is with the init container as soon as you list the pods. if you don't use the loop and the command fails then we don't see this.

Reply:
Actually I think when we tried without loop, we did it upstream way https://github.com/kubernetes/ingress-nginx/pull/1960/files
so didn't also use errexit or xtrace, which made init container pass and failures silent (not even logged);
I think we can drop the loop, but need to keep the two bash options, if not also set nounset option.